File: atividade-4/atividade.go
package main

import "fmt"

type Animal interface {
	Som() string
}

type Cachorro struct{}
type Gato struct{}

func (c Cachorro) Som() string {
	return "Latido"
}

func (g Gato) Som() string {
	return "Miau"
}

func main() {
	var animal Animal

	animal = Cachorro{}
	fmt.Println(animal.Som()) 

	animal = Gato{}
	fmt.Println(animal.Som()) 
}

File: atividade-4/atividade.py
class Animal:
    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return "Latido"

class Gato(Animal):
    def som(self):
        return "Miau"

cachorro = Cachorro()
gato = Gato()

print(cachorro.som()) 
print(gato.som())